File: clang/test/SemaCXX/pragma-pack.cpp
// RUN: %clang_cc1 -triple i686-apple-darwin9 -fsyntax-only -verify %s

namespace rdar8745206 {

struct Base {
  int i;
};

#pragma pack(1)
struct Sub : public Base {
  char c;
};

int check[sizeof(Sub) == 5 ? 1 : -1];

}

File: clang/test/SemaObjC/pragma-pack.m
// RUN: %clang_cc1 -triple i686-apple-darwin9 -fsyntax-only -verify -Wno-objc-root-class %s

// Make sure pragma pack works inside ObjC methods.  <rdar://problem/10893316>
@interface X
@end
@implementation X
- (void)Y {
#pragma pack(push, 1)
  struct x {
    char a;
    int b;
  };
#pragma pack(pop)
  typedef char check_[sizeof (struct x) == 5 ? 1 : -1];
}
@end
